doc/userguide/capture-hardware/dpdk.rst

```diff
@@ -184,3 +184,53 @@ Below is a sample configuration that demonstrates how to enable the interrupt mo
         - interface: 0000:3b:00.0
           interrupt-mode: true
           threads: 4
+
+.. _dpdk-automatic-interface-configuration:
+
+Automatic interface configuration
+---------------------------------
+
+A number of interface properties can be manually configured. However, Suricata
+can automatically configure the interface properties based on the NIC
+capabilities. This can be done by setting ``auto`` to ``mempool-size``,
+``mempool-cache-size``, ``rx-descriptors``, and ``tx-descriptors`` interface
+node properties.
+This will allow Suricata to automatically set the sizes of individual properties
+according to the best-effort calculation based on the NIC capabilities.
+For example, Rx/Tx descriptors are calculated based on the maximal "power of 2"
+that is lower or equal to the number of descriptors supported by the NIC.
+The number of mempool and its cache is then derived from these.
+
+Rx/Tx descriptors are set to the highest possible value to allow more buffer
+room when traffic spikes occur. However, it requires more memory.
+Individual properties can still be set manually if needed.
+
+.. note:: Mellanox ConnectX-4 NICs does not support auto-configuration of
```

Comment:
Need some more guidance here, as it's not yet working.

If I use the values from the `default` section, it works:
```
      mempool-size: 65535 # autocalculated based on Rx/Tx descriptors and threads
      mempool-cache-size: 257 # autocalculated from the mempool size
      rx-descriptors: 1024 # max number of descriptors
      tx-descriptors: 1024 # max number of descriptors
```

Reply:
Without it, I got
```
mlx5_net: No available register for sampler.
mlx5_net: No available register for sampler.
TELEMETRY: No legacy callbacks, legacy socket not created
Info: dpdk: 0000:01:00.0: creating 7 packet mempools of size 32767, cache size 217, mbuf size 2176 [DeviceConfigureQueues:runmode-dpdk.c:1413]
Error: dpdk: 0000:01:00.0: rte_pktmbuf_pool_create failed with code 12 (mempool: mp_6_0000:01:00.0) - Cannot allocate memory [DeviceConfigureQueues:runmode-dpdk.c:1422]
Error: dpdk: 0000:01:00.0: memory allocation failed - consider increasing the number of hugepages or freeing up some memory. [ParseDpdkConfigAndConfigureDevice:runmode-dpdk.c:1766]
```

Reply:
you would just need to define tx-descriptors: 16384 so:

```
      mempool-size: auto # autocalculated based on Rx/Tx descriptors and threads
      mempool-cache-size: auto # autocalculated from the mempool size
      rx-descriptors: auto
      tx-descriptors: 16384 # max number of descriptors
```

Reply:
Didn't work, got the same error then.